comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -21,40 +21,190 @@ public static bool IsDirectory(string path)
 			return File.GetAttributes(path).HasFlag(FileAttributes.Directory);
 		}
 
-		// Return the total size, in bytes, of all the files below the given directory
-		public static long GetTotalSize(string directory)
-		{
-			throw new NotImplementedException();
-		}
+        // Return the total size, in bytes, of all the files below the given directory
+        public static long GetTotalSize(string directory)
+        {
+            long size = 0;
+            Stack<string> searchStack = new Stack<string>();
+            searchStack.Push(directory);
+            while (searchStack.Count != 0)
+            {
+                var n = searchStack.Pop();
+                Console.WriteLine(n);
+                if (IsDirectory(n))
+                {
+                    var directories = Directory.GetDirectories(n);
+                    var files = Directory.GetFiles(n);
+                    foreach (string dir in directories)
+                    {
+                        searchStack.Push(dir);
+                    }
```

Comment:
This depth-first search code basically repeats in all of your functions -- could you have factored it out into a helper function so that it only appears once?